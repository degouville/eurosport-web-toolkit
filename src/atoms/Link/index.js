import React from 'react';
import styled from 'react-emotion';
import PropTypes from 'prop-types';

const Chevron = props => (
  <svg xmlns="http://www.w3.org/2000/svg" width="6" height="9" {...props}>
    <path fillRule="nonzero" d="M1.145 0L0 1.061 3.71 4.5 0 7.939 1.145 9 6 4.5z" />
  </svg>
);

const StyledChevron = styled(Chevron)`
  margin-left: 6px;

  path {
    fill: currentColor;
  }
`;

const LinkComponent = styled.a`
  color: ${({ theme }) => theme.link.color};
  &:focus,
  &:visited,
  &:hover {
    color: ${({ theme }) => theme.link.color};
  }
  text-decoration: none;
  text-align: center;
  text-transform: uppercase;
  display: block;
  font-size: 14px;
  font-weight: bold;
  letter-spacing: 1.27px;
  line-height: 24px;
`;

const Link = ({ children, ...props }) => (
  <LinkComponent {...props}>
    {children}
    <StyledChevron />
  </LinkComponent>
);

Link.displayName = 'Link';

Link.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Link;
