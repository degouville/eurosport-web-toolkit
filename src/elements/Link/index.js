import React from 'react';
import PropTypes from 'prop-types';

const Link = ({ linkComponent: LinkComponent, ...props }) => <LinkComponent {...props} />;

Link.propTypes = {
  linkComponent: PropTypes.func,
};

Link.defaultProps = {
  linkComponent: ({ href, children, ...props }) => (
    <a href={href} {...props}>
      {children}
    </a>
  ),
};

export default Link;
