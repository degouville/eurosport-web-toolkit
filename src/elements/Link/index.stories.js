import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';
import { Link as OverridableLink } from '../..';

const indexStories = storiesOf('Elements|Link', module);

indexStories
  .add(
    'default Link',
    withInfo()(() => <OverridableLink href={text('href', 'goo.gl')}>{text('Label', 'Default Link')}</OverridableLink>)
  )
  .add(
    'custom Link span',
    withInfo()(() => (
      <OverridableLink linkComponent={props => <span {...props} />}>{text('Label', 'Custom Link')}</OverridableLink>
    ))
  );
