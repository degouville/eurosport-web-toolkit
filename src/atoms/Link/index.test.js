import React from 'react';
import { shallow } from 'enzyme';
import Link from '.';
import { theme } from '../..';

it('renders a Link', () => {
  expect(
    shallow(
      <Link theme={theme} href="/">
        Click me !
      </Link>
    )
  ).toMatchSnapshot();
});
