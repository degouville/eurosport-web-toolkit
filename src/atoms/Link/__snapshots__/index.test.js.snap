// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Link 1`] = `
<Styled(a)
  href="/"
  theme={
    Object {
      "button": Object {
        "primary": Object {
          "backgroundColor": "#d50032",
          "color": "white",
        },
        "secondary": Object {
          "color": "#64dcff",
        },
      },
      "link": Object {
        "color": "#64dcff",
      },
    }
  }
>
  Click me !
  <Styled(Chevron) />
</Styled(a)>
`;
